cpp/include/cudf/copying.hpp

```diff
@@ -506,41 +506,40 @@ std::unique_ptr<column> copy_if_else(column_view const& lhs, column_view const&
                                     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
 
 /**
-* @brief Creates a new table by shifting all values in all columns by an offset.
+* @brief Creates a new column by shifting all values by an offset.
 *
-* Elements will be determined by `output[col][idx] = input[col][idx - offset]`.
+* Elements will be determined by `output[idx] = input[idx - offset]`.
 * Some elements in the output may be indeterminable from the input. For those
 * elements, the value will be determined by `fill_values`.
 *
 * Examples
 * -------------------------------------------------
-* input      = { [0, 1, 2, 3, 4], [5, 4, 3, 2, 1] }
-* offset     = 3
-* fill_values = { @, 7 }
-* return     = { [@, @, @, 0, 1], [7, 7, 7, 5, 4] }
+* input       = [0, 1, 2, 3, 4]
+* offset      = 3
+* fill_values = @
+* return      = [@, @, @, 0, 1]
 * -------------------------------------------------
-* input      = { [0, 1, 2, 3, 4], [5, 4, 3, 2, 1] }
-* offset     = -2
-* fill_values = { 2, @ }
-* return     = { [2, 3, 4, 2, 2], [3, 2, 1, @, @] }
+* input       = [5, 4, 3, 2, 1]
+* offset      = -2
+* fill_values = 7
+* return      = [3, 2, 1, 7, 7]
 *
-* @note if a column is nullable, it's output column will be nullable.
-* @note if a fill value is null, it's output column will be nullable.
+* @note if the input is nullable, the output will be nullable.
+* @note if the fill value is null, the output will be nullable.
 *
```

Comment:
Would it make sense to change this so that if there are no nulls, the output will not be nullable?

Reply:
If the consumer doesn't want nulls in the result, they can pass a view with no nulls + a non-null fill value. If we just check for the null count, we take that choice away and potentially do extra work (if the null count isn't already computed).